features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -1225,30 +1222,41 @@ void ATHandler::flush()
     }
 }
 
-void ATHandler::debug_print(const char *p, int len)
+void ATHandler::debug_print(const char *p, int len, ATType type)
 {
 #if MBED_CONF_CELLULAR_DEBUG_AT
     if (_debug_on) {
-#if MBED_CONF_MBED_TRACE_ENABLE
-        mbed_cellular_trace::mutex_wait();
-#endif
-        char c;
+        const int buf_size = len * 4; // x4 -> reserve space for extra characters
```

Comment:
`len * 4 + 1` where +1 for sprintf zero termination.

Reply:
Fixed